core/src/locations/installation.cpp

```diff
@@ -0,0 +1,100 @@
+#include "installation.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QFile>
+#include <QDir>
+
+#include "customexceptions.h"
+#include "config.h"
+#include "ioutils.h"
+
+static const QString patcherInstallationLocationKey = "patcherInstallationLocation";
+static const QString appInstallationLocationKey = "appInstallationLocation";
```

Comment:
Wouldn't it be simpler to store just installation location and later generate both patcher & app installation dirs from that?

This would decrease complexity:
- no need to use JSON and parsing, simply grab location from text file
- it's always guaranteed that both patcher and app are placed inside the same directory, as we always do (I can't think about situation where we wouldn't want to do that)

Reply:
It would be simpler, but not by a lot. And having this in JSON makes it more extensible. We already have a lot of these files with a single line of text or something and that's getting slightly difficult to manage. 

I can't for now think of a situation when the patcher or the app would be installed in a different location but you never now - I'm giving us more options in the future.